src/backend/distributed/executor/multi_utility.c

```diff
@@ -395,7 +395,11 @@ multi_ProcessUtility(Node *parsetree,
 	standard_ProcessUtility(parsetree, queryString, context,
 							params, dest, completionTag);
 
-	PostProcessUtility(parsetree);
+	/* don't run post-process code for local commands */
+	if (ddlJobs != NIL)
+	{
+		PostProcessUtility(parsetree);
```

Comment:
Expand the surrounding context and you'll see that this area potentially runs as a different user. Presently that's only in the `COPY` case, I believe, so it isn't important, but it seemed to make sense to have the "local callback" code run as that user as well if other local commands did.

I can move it, but when initially writing it it seemed more "correct" to have all local logic run as the special user. If it's unimportant to you (or whoever wants to review this), no biggie.